source/byte_buf.c

```diff
@@ -1633,3 +1633,46 @@ bool aws_isspace(uint8_t ch) {
             return false;
     }
 }
+
+static int s_read_unsigned(struct aws_byte_cursor cursor, uint64_t *dst, uint8_t base) {
+    uint64_t val = 0;
+    *dst = 0;
+
+    if (cursor.len == 0) {
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    const uint8_t *hex_to_num_table = aws_lookup_table_hex_to_num_get();
+
+    /* read from left to right */
+    for (size_t i = 0; i < cursor.len; ++i) {
+        const uint8_t c = cursor.ptr[i];
+        const uint8_t cval = hex_to_num_table[c];
+        if (cval >= base) {
+            return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        }
+
+        const uint64_t prev_val = val;
+
+        val *= base;
+        if (val < prev_val) {
```

Comment:
i think we have functions for this. something like aws_mul_overflow_check()

Reply:
Changed to `aws_mul_u64_checked` and `aws_add_u64_checked` , thanks a lot!

Reply:
yeah, use `aws_add_u64_checked()` and `aws_mul_u64_checked()` instead of my hacky overflow checks. I don't think those functions existed when I first wrote this?